File: 1-Observer/python/main.py
from observers import ObserverA
from observers import ObserverB
from observers import ObserverC
from subject import Subject

observer1 = ObserverA()
observer2 = ObserverB()
observer3 = ObserverC()

subject = Subject()

subject.subscribe(observer1)
subject.subscribe(observer2)
subject.subscribe(observer3)

subject.update()

File: 1-Observer/c++/src/main.cpp
#include <iostream>
#include <observer.h>
#include <subject.h>
#include <vector>

using namespace std;

int main(){

    ObserverA* observerA = new ObserverA();
    ObserverB* observerB = new ObserverB();
    ObserverC* observerC = new ObserverC();

    Subject* subject = new Subject();
    subject->subscribe(observerA);
    subject->subscribe(observerB);
    subject->subscribe(observerC);

    subject->update();

    return 0;
}